db/common_test.go

```diff
@@ -0,0 +1,170 @@
+// Package db - common test
+// The following file contains all the common functions for testing.
+package db
+
+import (
+	"context"
+	"crypto/rand"
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"github.com/packethost/pkg/log"
+)
+
+func newCTX() ctx {
+	c, closer := context.WithCancel(context.Background())
+	return ctx{
+		context: c,
+		cancel:  closer,
+	}
+}
+
+func (a *testArgs) setCTX() {
+	a.ctx = newCTX()
+}
+
+func (a *testArgs) setLogger() {
+	var err error
+	if a.logger, err = log.Init("github.com/tinkerbell/tink"); err != nil {
+		panic(err)
```

Comment:
t.Fail instead of panic would be better.

Reply:
Will add.